http_server.py

```diff
@@ -0,0 +1,75 @@
+import socket
+#  from urllib2 import HTTPError
+
+ADDR = ("127.0.0.1", 8000)
+server = socket.socket(
+    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+)
+server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+
+def parse_request(request):
+    try:
+        request = request.split("\r\n\r\n", 1)
+        req = request[0].split("\r\n")
+        for i, r in enumerate(req):
+            req[i] = r.split()
+        method = req[0][0].upper()
+        uri = req[0][1]
+        proto = req[0][2].upper()
+        headers = {}
+        for line in req[1:]:
+            headers[line[0].upper()] = line[1:]
+        if method == "GET":
+            if proto == "HTTP/1.1" and "HOST:" in headers:
+                return uri
+            else:
+                raise SyntaxError("400 Bad Request")
+        else:
+            raise ValueError("405 Method Not Allowed")
```

Comment:
these sorts of nested if/else clauses are a bit hard to read.  I'd rather see a series of one-level `if` statements.